src/routers/__tests__/SwitchRouter-test.js

```diff
@@ -78,6 +81,75 @@ describe('SwitchRouter', () => {
     expect(state3.index).toEqual(0);
   });
 
+  test('handles order backBehavior', () => {
+    const routerHelper = new ExampleRouterHelper({ backBehavior: 'order' });
+    expect(routerHelper.getCurrentState().routeKeyHistory).toBeUndefined();
+
+    expect(
+      routerHelper.applyAction({
+        type: NavigationActions.NAVIGATE,
+        routeName: 'C',
+      })
+    ).toMatchObject({ index: 2 });
+
+    expect(
+      routerHelper.applyAction({ type: NavigationActions.BACK })
+    ).toMatchObject({ index: 1 });
+
+    expect(
+      routerHelper.applyAction({ type: NavigationActions.BACK })
+    ).toMatchObject({ index: 0 });
+
+    expect(
+      routerHelper.applyAction({ type: NavigationActions.BACK })
+    ).toMatchObject({ index: 0 });
+  });
+
+  test('handles history backBehavior', () => {
+    const routerHelper = new ExampleRouterHelper({ backBehavior: 'history' });
+    expect(routerHelper.getCurrentState().routeKeyHistory).toMatchObject(['A']);
+
+    expect(
+      routerHelper.applyAction({
+        type: NavigationActions.NAVIGATE,
+        routeName: 'B',
+      })
+    ).toMatchObject({ index: 1, routeKeyHistory: ['A', 'B'] });
+
+    expect(
+      routerHelper.applyAction({
+        type: NavigationActions.NAVIGATE,
+        routeName: 'A',
+      })
+    ).toMatchObject({ index: 0, routeKeyHistory: ['B', 'A'] });
+
+    expect(
+      routerHelper.applyAction({
+        type: NavigationActions.NAVIGATE,
+        routeName: 'C',
+      })
+    ).toMatchObject({ index: 2, routeKeyHistory: ['B', 'A', 'C'] });
```

Comment:
just a dev note: here the object is matched "partially" but the object is matched "fully". Not really explicit from jest doc though...

If the state contains `['B', 'A', 'C', 'D']` it will effectively fail because of presence of D

Reply:
Makes sense that its a shallow match 👌